lib/df-detection.c

```diff
@@ -0,0 +1,209 @@
+#include <linux/df-detection.h>
+#include <linux/debugfs.h>
+#include <linux/fs.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/random.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+#include <linux/stacktrace.h>
+#include <linux/sys.h>
+#include <linux/types.h>
+
+void dfetch_add_address(const void *addr, size_t len, unsigned long caller,
+		 void *kernel_addr)
+{
+	if (!in_task() || current->dfetch_addresses == NULL || current->dfetch_pairs == NULL ||
+	    addr > TASK_SIZE)
```

Comment:
Nit: you could move this `addr > TASK_SIZE` check to the following if-statement, so that two `addr` checks belong together. This will also save you a line.